FirebaseRemoteConfig/Sources/RCNPersonalization.m

```diff
@@ -37,16 +38,35 @@ - (void)logArmActive:(NSString *)key config:(NSDictionary *)config {
   }
 
   NSDictionary *metadata = ids[key];
-  if (!metadata || metadata[kPersonalizationId] == nil) {
+  if (!metadata) {
     return;
   }
 
+  NSString *choiceId = metadata[kChoiceId];
+  if (choiceId == nil) {
+    return;
+  }
+
+  // This gets dispatched to a serial queue, so this is OK. But even if not, it'll just possibly
+  // log more.
+  if (self->_armsCache[key] == choiceId) {
+    return;
+  }
+  self->_armsCache[key] = choiceId;
+
   [self->_analytics logEventWithOrigin:kAnalyticsOriginPersonalization
                                   name:kAnalyticsPullEvent
                             parameters:@{
-                              kArmKey : metadata[kPersonalizationId],
-                              kArmValue : values[key].stringValue
+                              kArmKey : key,
+                              kArmValue : values[key].stringValue,
+                              kPersonalizationIdLogKey : metadata[kPersonalizationId],
+                              kArmIndexLogKey : metadata[kArmIndex],
+                              kGroup : metadata[kGroup]
                             }];
+
+  [self->_analytics logEventWithOrigin:kAnalyticsOriginPersonalization
+                                  name:kAnalyticsPullEventInternal
+                            parameters:@{kChoiceIdLogKey : choiceId}];
```

Comment:
Thinking: the [design](http://go/p13n-choice-metadata) lists one thing we want to log in the internal event:

1. Choice ID, satisfied by the `_fpid` param

Thinking: the `_fpc` event name is correct per the [events proposal](http://go/p13n-events).